frontends/web/src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,133 @@
+
+/**
+ * Copyright 2024 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReactNode } from 'react';
+import Select, {
+  components,
+  SingleValueProps,
+  OptionProps,
+  DropdownIndicatorProps,
+  MultiValueProps,
+  Props as ReactSelectProps,
+  ActionMeta,
+} from 'react-select';
+import styles from './dropdown.module.css';
+
+export type TOption<T = any> = {
+  label: string;
+  value: T;
+};
+
+type SelectProps<T = any, IsMulti extends boolean = false> = Omit<
+  ReactSelectProps<TOption<T>>,
+  'onChange' | 'isMulti'
+> & {
+  renderOptions: (selectedItem: TOption<T>) => ReactNode; // Function to render options and selected value for single dropdown
+  onChange: (
+    newValue: IsMulti extends true ? TOption<T>[] : TOption<T>,
+    actionMeta: ActionMeta<TOption<T>>
+  ) => void;
+  isMulti?: IsMulti;
+};
+
+const DropdownIndicator = (props: DropdownIndicatorProps<TOption>) => (
+  <components.DropdownIndicator {...props}>
+    <div className={styles.dropdown} />
+  </components.DropdownIndicator>
+);
+
+const SelectSingleValue = (props: SingleValueProps<TOption> & { selectProps: any }) => {
+  const { renderOptions } = props.selectProps;
+
+  return (
+    <components.SingleValue {...props}>
+      {renderOptions(props.data)}
+    </components.SingleValue>
+  );
+};
+
+const Option = (props: OptionProps<TOption> & { selectProps: any }) => {
+  const { renderOptions } = props.selectProps;
+  return (
+    <components.Option {...props}>
+      {renderOptions(props.data)}
+    </components.Option>
+  );
+};
+
+// Displays up to 3 most recently selected items, separated by commas.
+// Truncates with '...' if more than 3 are selected.
+const CustomMultiValue = ({ index, getValue }: MultiValueProps<TOption>) => {
+  // displays only the first MultiValue instance
+  if (index > 0) {
+    return null;
+  }
+
+  const selectedValues = getValue();
+  const maxVisible = 3;
+
+  const displayedValues = selectedValues
+    .slice()
+    .reverse() // shows the latest selected values first
+    .slice(0, maxVisible)
+    .map((option) => option.value)
+    .join(', ');
```

Comment:
I tried to find the old logic that shows 3 most recent selected items could you point me to it?